evm_trace/base.py

```diff
@@ -54,3 +57,121 @@ class CallTreeNode(BaseModel):
     @validator("address", "calldata", "returndata", pre=True)
     def validate_hexbytes(cls, v) -> HexBytes:
         return _convert_hexbytes(cls, v)
+
+
+def get_calltree_from_trace(
+    trace: Iterator[TraceFrame], show_internal=False, **root_node_kwargs
+) -> CallTreeNode:
+    """
+    Creates a CallTreeNode from a given transaction trace.
+
+    Args:
+        trace (Iterator[TraceFrame]): Iterator of transaction trace frames.
+        show_internal (bool): Boolean whether to display internal calls. Defaulted to False.
+        root_node_kwargs (dict): Keyword argments passed to the root ``CallTreeNode``.
+
+    Returns:
+        :class:`~evm_trace.base.CallTreeNode: Call tree of transaction trace.
+    """
+
+    return _create_node_from_call(
+        trace=trace,
+        show_internal=show_internal,
+        **root_node_kwargs,
+    )
+
+
+def _extract_memory(offset: HexBytes, size: HexBytes, memory: List[HexBytes]) -> HexBytes:
+    """
```

Comment:
In general, you don't need "proper" documentation for methods prefixed with `_` (internal or private methods).
It is sometimes nice to know what it's doing, but always feel free to omit or be more brief